src/Gaufrette/Adapter/GridFS.php

```diff
@@ -18,41 +19,52 @@ class GridFS implements Adapter,
                         MetadataSupporter,
                         ListKeysAware
 {
-    private $metadata = array();
-    protected $gridFS = null;
+    /** @var array */
+    private $metadata = [];
+
+    /** @var Bucket */
+    private $bucket;
 
     /**
-     * @param \MongoGridFS $gridFS
+     * @param Bucket $bucket
      */
-    public function __construct(MongoGridFs $gridFS)
+    public function __construct(Bucket $bucket)
     {
-        $this->gridFS = $gridFS;
+        $this->bucket = $bucket;
     }
 
     /**
      * {@inheritdoc}
      */
     public function read($key)
     {
-        $file = $this->find($key);
+        try {
+            $stream = $this->bucket->openDownloadStreamByName($key);
+        } catch (FileNotFoundException $e) {
+            return false;
+        }
 
-        return ($file) ? $file->getBytes() : false;
+        try {
+            return stream_get_contents($stream);
+        } finally {
+            fclose($stream);
+        }
     }
 
     /**
      * {@inheritdoc}
      */
     public function write($key, $content)
     {
-        if ($this->exists($key)) {
-            $this->delete($key);
-        }
+        $stream = $this->bucket->openUploadStream($key, ['metadata' => $this->getMetadata($key)]);
 
-        $metadata = array_replace_recursive(array('date' => new MongoDate()), $this->getMetadata($key), array('filename' => $key));
-        $id = $this->gridFS->storeBytes($content, $metadata);
-        $file = $this->gridFS->findOne(array('_id' => $id));
+        try {
+            fwrite($stream, $content);
+        } finally {
+            fclose($stream);
+        }
 
-        return $file->getSize();
+        return strlen($content);
```

Comment:
We should return the result of `fwrite($stream, $content)` instead because with this implementation there is no way to even know an error occurred.